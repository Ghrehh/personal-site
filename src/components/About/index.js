import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.module.css';

class About extends React.Component {
  renderAbout() {
    return this.props.projects.map((project, index) => (
      <section
        className={styles.project}
        key={index}
        style={{ backgroundColor: project.backgroundColor }}
      >
        <h2 className={styles.name}>{project.name}</h2>
        <p className={styles.description}>{project.description}</p>
      </section>
    ));
  }

  render() {
    return (
      <article>
        <header>
          <h2 className={styles.header}>PROJECTS</h2>
          <div className={styles.headerUnderline} />
        </header>

        <div className={styles.projects}>{this.renderAbout()}</div>
      </article>
    );
  }
}

About.propTypes = {
  projects: PropTypes.array.isRequired
};

export default About;
