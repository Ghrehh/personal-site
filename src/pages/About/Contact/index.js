import React from 'react';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faCoffee from '@fortawesome/fontawesome-free-solid/faCoffee';

import styles from './styles.module.css';

const Contact = () => (
  <section>
    <h1 className={styles.contactTitle}>CONTACT</h1>
    <FontAwesomeIcon icon={faCoffee} />
  </section>
);

export default Contact;
