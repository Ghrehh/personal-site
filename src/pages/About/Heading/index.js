// @flow
import React from 'react';

import * as childrenAndClassName from 'models/children-and-class-name';
import styles from './styles.module.css';

const Heading = (props: childrenAndClassName.Type) => {
  const combinedClass = `${styles.heading} ${props.className}`;

  return <h1 className={combinedClass}>{props.children}</h1>;
};

Heading.defaultProps = childrenAndClassName.defaultProps;

export default Heading;
