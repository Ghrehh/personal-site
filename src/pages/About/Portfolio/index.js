// @flow
import React from 'react';

import SectionHeading from 'components/SectionHeading';
import Button from 'components/Button';
import LinkButton from 'components/LinkButton';
import ButtonContainer from 'components/ButtonContainer';
import * as pathConstants from 'constants/paths';
import styles from './styles.module.css';

const Portfolio = () => (
  <section className={styles.display}>
    <div className={styles.displayInner}>
      <SectionHeading className={styles.heading}>PORTFOLIO</SectionHeading>

      <p className={styles.body}>
        I open-source most of my personal projects on GitHub. Why not check them
        out?
      </p>

      <ButtonContainer>
        <LinkButton link={pathConstants.PROJECTS}>View Portfolio</LinkButton>
        <Button className={styles.gitHubButton}>GitHub</Button>
      </ButtonContainer>
    </div>
  </section>
);

export default Portfolio;
