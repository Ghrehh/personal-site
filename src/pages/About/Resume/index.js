// @flow
import React from 'react';

import SectionHeading from 'components/SectionHeading';
import ButtonContainer from 'components/ButtonContainer';
import Button from 'components/Button';
import styles from './styles.module.css';

const buttonStyle = additionalStyle => `${styles.button} ${additionalStyle}`;

const Resume = () => (
  <section className={styles.resume}>
    <div className={styles.resumeInner}>
      <SectionHeading className={styles.heading}>CV/RESUME</SectionHeading>

      <p className={styles.body}>
        Please view my CV for a detailed description of my past and current
        work.
      </p>

      <ButtonContainer className={styles.buttonContainer}>
        <Button className={buttonStyle(styles.viewButton)}>View CV</Button>

        <Button className={buttonStyle(styles.downloadButton)}>
          Download CV
        </Button>
      </ButtonContainer>
    </div>
  </section>
);

export default Resume;
