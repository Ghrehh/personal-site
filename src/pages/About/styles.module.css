.intro {
  padding: 50px 20px 65px;
}

.heading {
  color: var(--red);
}

.subHeading {
  font-size: 24px;
  text-align: center;
  font-weight: 800;
  letter-spacing: 2px;
  margin: 0 auto 1px;
  max-width: 600px;
  line-height: 28px;
  font-family: var(--slab-font);
  color: var(--blue);
}

.body {
  font-size: 20px;
  text-align: center;
  font-weight: 100;
  line-height: 30px;
  max-width: 470px;
  margin: auto;
  color: var(--blue);
}
