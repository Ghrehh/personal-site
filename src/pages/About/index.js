import React from 'react';

import Page from 'components/Page';
import Heading from 'components/Heading';
import SubHeading from 'components/SubHeading';
import Portfolio from './Portfolio';
import Contact from './Contact';

import styles from './styles.module.css';

const About = () => (
  <Page>
    <section className={styles.intro}>
      <Heading className={styles.heading}>HEY, I'M GERARD BURNS</Heading>

      <SubHeading className={styles.subHeading}>
        A SOFTWARE ENGINEER FROM BELFAST
      </SubHeading>

      <p className={styles.body}>
        with experience working with both front-end and back-end technologies,
        primarily React, Ruby on Rails, Django and Vue.
      </p>
    </section>

    <Portfolio />
    <Contact />
  </Page>
);

export default About;
