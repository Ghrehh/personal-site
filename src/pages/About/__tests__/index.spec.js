import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import About from '../';

describe('About', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};
    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(<About />);
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
