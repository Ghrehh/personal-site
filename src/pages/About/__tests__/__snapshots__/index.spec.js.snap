// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders the component correctly 1`] = `
<Page>
  <section
    className="intro"
  >
    <PageHeading
      className="heading"
    >
      HEY, I'M GERARD BURNS
    </PageHeading>
    <h2
      className="subHeading"
    >
      A SOFTWARE ENGINEER FROM BELFAST
    </h2>
    <p
      className="body"
    >
      with experience working with both front-end and back-end technologies, primarily React, Ruby on Rails, Django and Vue.
    </p>
  </section>
  <Portfolio />
  <Resume />
</Page>
`;
